contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3322,6 +3322,38 @@ $$
 $$
 LANGUAGE sql PARALLEL SAFE STABLE;
 
+CREATE OR REPLACE FUNCTION OBJECT_DEFINITION(IN object_id INT)
+RETURNS sys.NVARCHAR(4000)
+AS $$
+DECLARE
+    type char(2);
+BEGIN
+
+    IF NOT EXISTS(SELECT ao.object_id FROM sys.all_objects ao WHERE ao.object_id = $1)
```

Comment:
Can we save this `all_objects` view lookup? I am thinking of directly searching the OID as follows:

1. Search in `sys.check_constraints` return if we get non-null result otherwise,
2. Search in `sys.default_constraints` return if we get non-null result otherwise,
3. Search in `sys.all_sql_modules` return if we get non-null result otherwise,
4. return null
I am asking this since `sys.all_objects` lookup can be quite expensive.